### PR TITLE
Implement notification on denial (CO-1658)

### PR DESCRIPTION
**Pull request by lionick**
DB CHANGES:
```
alter table cm_co_enrollment_flows
	add denial_template_id int;

alter table cm_co_petitions
	add approver_comment text;

alter table cm_co_enrollment_flows
	add constraint cm_co_enrollment_flows_denial_template_id_fkey
		foreign key (denial_template_id) references cm_co_message_templates (id);
```

### CODE REVIEW

File: app/Config/Schema/schema.xml [hunk 1]
```diff
@@ -1574,6 +1577,7 @@
     <field name="petitioner_token" type="C" size="48" />
     <field name="enrollee_token" type="C" size="48" />
     <field name="return_url" type="C" size="256" />
+    <field name="approver_comment" type="C" size="256" />
```
Comment (ioigoume) [inline app/Config/Schema/schema.xml]: let's make this type `text`

File: app/Model/CoPetition.php [hunk 2]
```diff
@@ -2033,8 +2033,11 @@ public function sendApprovalNotification($id, $actorCoPersonId, $action='approva
     
     $args = array();
     $args['conditions']['CoPetition.id'] = $id;
-    $args['contain']['CoEnrollmentFlow'] = 'CoEnrollmentFlowApprovalMessageTemplate';
-    $args['contain']['CoEnrollmentFlow'] = 'CoEnrollmentFlowFinMessageTemplate';
+     /* $args['contain']['CoEnrollmentFlow'] = array(
+      'CoEnrollmentFlowAppMessageTemplate',
+      'CoEnrollmentFlowDenMessageTemplate',
+      'CoEnrollmentFlowFinMessageTemplate'
```
Comment (ioigoume) [inline app/Model/CoPetition.php]: Do we need the comments?

File: app/View/Elements/petitionActions.ctp [hunk 3]
```diff
@@ -88,42 +88,53 @@ if($permissions['isEnrollee'] && !empty($vv_coef_next_step) && !in_array($status
...
+                                'name'  => 'action')
     );
   }
+  print $this->Form->submit(_txt('op.deny'),
+                            array(
+                              'class' => 'cancelbutton deny-button',
+                              'name'  => 'action')
+  );
+
+  print $this->Form->input('approver_comment',
```
Comment (ioigoume) [inline app/View/Elements/petitionActions.ctp]: Let's make this a textarea

**Review by ioigoume — CHANGES_REQUESTED** [review]

File: app/Config/Schema/schema.xml (hunk 1 above)
Comment (lionick) [reply app/Config/Schema/schema.xml]: done

File: app/Model/CoPetition.php (hunk 2 above)
Comment (lionick) [reply app/Model/CoPetition.php]: nope

File: app/View/Elements/petitionActions.ctp (hunk 3 above)
Comment (lionick) [reply app/View/Elements/petitionActions.ctp]: done

**Review by ioigoume — APPROVED** [review]
LGTM

**Pull request merged**